### ISSUE TITLE
Feature request: components should be able to listen for changes in rendered stylesheet and matching media queries

### ISSUE THREAD

**Issue by david-davidson** [FormidableLabs/radium#523]
Now that Radium uses `<style>` tags to handle media queries, it would be useful if a component that implements media queries could subscribe to changes in the rendered `StyleSheet`, as well as changes in which media queries are currently active (matching).

Here's one use case: suppose we have a component that 1.) implements media queries defining its width in _percentages_ and 2.) needs to somehow track its width in _pixels_, maybe to pass as props to a child component. As we navigate to the that component's route, its media queries are added to the `StyleSheet`, which re-renders asynchronously via `setState`. Only after the new CSS is flushed to the DOM does the component's width change; this is when we'd want to do something like fire a callback that tracks `offsetWidth` via `findDOMNode`. For the same reason, as the viewport changes width, it'd be nice to have a way to be notified when the matching media queries have changed.

I'm imagining a subscription API similar to what's exposed by `StyleKeeper`, where `StyleSheet` would call its subscribers in `componentDidUpdate`.

@ianobermiller, does this sound like a reasonable feature and approach?

**Comment by ianobermiller** [comment]
While this is an interesting idea, I don't think you need to add anything to Radium to accomplish it. You need a "resize detector" for an element, there are various ways to implement it, and I've done so in one way in this jsbin: https://jsbin.com/mokixo/edit?js,output

I believe a more performant solution would use an invisible iframe and requestAnimationFrame.

**Comment by alexlande** [comment]
Seems like this one is fairly settled. Please re-open if it needs more discussion :soccer:
